docs/multigpu.md

```diff
@@ -1,62 +1,83 @@
+`run_on_main` ensures that a specific function is executed only once, in only one process, forcing other processes to wait. It is frequently used to run a dataset preparation step in recipes.
 
-Using DDP in speechbrain with a single server (node) is quite easy:
+Many functions like `Brain.fit` are written to be DDP-aware. In practice, there is not a lot you need to do to make your code DDP-safe, but it is something you should keep in mind.
 
-```
+> **NOTE:**
+> With DDP, batch size is defined for a single process/GPU. This is different from Data Parallel (DP), where batches are split according to the number of GPUs. For example, with DDP, if you specify a batch size of 16, each GPU/process will use batches of 16 regardless of how many GPUs you have.
+
+### Single-node setup
+
+_This covers the case where you want to split training across **multiple GPUs** on **a single machine** (node)._
+
+Using SpeechBrain, this would look like:
+
+```bash
 cd recipes/<dataset>/<task>/
-python -m torch.distributed.launch --nproc_per_node=4 experiment.py hyperparams.yaml --distributed_launch --distributed_backend='nccl'
+python -m torch.distributed.launch --nproc_per_node=4 experiment.py hyperparams.yaml --distributed_launch
 ```
 
-Where:
-- nproc_per_node must be equal to the number of GPUs.
-- distributed_backend is the type of backend managing multiple processes synchronizations (e.g, 'nccl', 'gloo'). Try to switch the DDP backend if you have issues with nccl.
+... where `nproc_per_node` is the the number of processes to spawn/GPUs to use.
+
+### Multi-node setup
+
+_This covers the case where you want to split training across **multiple machines** on a network, with any amount of GPUs per machine._
+
+Note that using DDP across multiple machines introduces a **communication overhead** that might slow down training significantly, sometimes more than if you were to train on a single node! This largely depends on the network speed between the nodes.
+Make sure you are actually observing any benefits from distributing the work across machines.
+
+While DDP is more efficient than `DataParallel`, it is somewhat prone to exhibit unexpected bugs. DDP is quite server-dependent, so some setups may face issues. If you are encountering problems, make sure PyTorch is well up to date.
 
-Running DDP over multiple servers (nodes) is quite system dependent. Let's start with a simple example where a user is able to connect to each node directly. If we want to run 2 GPUs on 2 different nodes (i.e total of 4 GPUs), we must do:
+#### Basics & manual multi-node setup
 
-```shell
+Let's start with a simple example where a user is able to connect to each node directly. If we want to run 2 GPUs on 2 different nodes (for a total of 4 GPUs), we must do:
```

Comment:
I think it would be great to remind the user that `--nproc_per_node=2` will expect 2 GPUs, hence, having 2 nodes will lead to 4 GPUs. I mean, you already said it but I think we should be more specific on that